src/integrations/GoogleCloudPubSubIntegration.ts

```diff
@@ -0,0 +1,336 @@
+        ThundraLogger.debug('<GoogleCloudPubSubIntegration> Wrap');
+
+        const integration = this;
+        function requestWrapper(original: any, wrappedFunctionName: string) {
+            integration.wrappedFuncs[wrappedFunctionName] = original;
+            return function internalWrapper(config: any, callback: any) {
+
+                const orginalCallback = callback;
+                let span: ThundraSpan;
+                let reachedToCallOriginalFunc: boolean = false;
+                try {
+
+                    ThundraLogger.debug('<GoogleCloudPubSubIntegration> Tracing GoogleCloudPubSubIntegration request:', config);
+
+                    const { tracer } = ExecutionContextManager.get();
+
+                    if (!tracer) {
+                        ThundraLogger.debug('<GoogleCloudPubSubIntegration> Skipped tracing request as no tracer is available');
+                        return original.apply(this, [config, orginalCallback]);
+                    }
+
+                    const originalFunction = integration.getOriginalFunction(wrappedFunctionName);
+                    const parentSpan = tracer.getActiveSpan();
+
+                    const topic = GooglePubSubUtils.getTopic(config);
+                    const operationName = `${config.method.toUpperCase()}-${topic}`;
+
+                    ThundraLogger.debug(`<GoogleCloudPubSubIntegration> Starting publish span with name ${operationName}`);
+
+                    span = tracer._startSpan(operationName, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.GOOGLEPUBSUB,
+                        disableActiveStart: true,
+                    });
+
+                    const getMessageBody = () => {
+                        if (!config || !config.reqOpts || !config.reqOpts.messages) {
+                            return;
+                        }
+
+                        return GooglePubSubUtils.parseMessages(config.reqOpts.messages);
+                    };
+
+                    const meesageContent = getMessageBody();
+
+                    const tags = {
+                        [GooglePubSubTags.PROJECT_ID]: this.projectId,
+                        [GooglePubSubTags.TOPIC_NAME]: topic,
+                        ...((meesageContent && !thundraConfig.maskGooglePublishRequestBody) ? {
+                            [GooglePubSubTags.MESSAGE]: JSON.stringify(getMessageBody()),
+                        } : undefined),
+                        [SpanTags.OPERATION_TYPE]: config.method
+                            ? (GooglePubSubOperationTypes[`${config.method.toUpperCase()}`] || config.method)
+                            : SpanTypes.GOOGLEPUBSUB,
+                        [SpanTags.SPAN_TYPE]: SpanTypes.GOOGLEPUBSUB,
+                        [SpanTags.TRACE_LINKS]: [span.spanContext.spanId],
+                        [SpanTags.TOPOLOGY_VERTEX]: true,
+                    };
```

Comment:
It might be better to tag resource names here 
```
[SpanTags.RESOURCE_NAMES]: [topic]
```
So the resource name will be assigned from here instead of operation name (which is default value for resource name).

And so we can link publisher and poller/subscriber sides without need to add `TriggerHeaderTags.RESOURCE_NAME` into message attributes. Note that In this case, poller/subscriber side must use topic name as trigger resource name

Reply:
The reason why I suggest this way is that resource names should not include operation types but only resource name itself. And here if we don't follow this approach, since we set the operation name with the operation method prefix, resource name includes operation method as prefix by default.